### PR TITLE
Add kafka as an output option

### PR DESCRIPTION
**Pull request by boernd**
Kafka as an additional output option. Tested with kafka 0.10.0.1

### CODE REVIEW

File: src/riemann/kafka.clj [hunk 1]
```diff
@@ -0,0 +1,30 @@
+(ns riemann.kafka
+  "Forwards events to Kafka."
+  (:require [kinsky.client :as client]))
+
+(defn kafka
+  "Returns a functions which accepts an event and sends it to a Kafka topic.
```
Comment (jamtur01) [inline src/riemann/kafka.clj]: `Returns a function that accepts an event and sends it to a Kafka topic.`

File: src/riemann/kafka.clj [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  ([] (kafka {}))
+  ([opts]
+  (let
+    [opts (merge {:bootstrap-servers "localhost:9092"
+                  :topic "riemann"
+                  :value-serializer client/json-serializer}
+                 opts)
+     producer (client/producer {:bootstrap.servers (:bootstrap-servers opts)}
+                 (:value-serializer opts))]
+    (fn[event] (client/send!
```
Comment (mcorbin) [inline src/riemann/kafka.clj]: Indentation problem, should be :
(fn [event]
  (...

File: src/riemann/kafka.clj [hunk 3]
```diff
@@ -0,0 +1,30 @@
+(ns riemann.kafka
+  "Forwards events to Kafka."
+  (:require [kinsky.client :as client]))
+
+(defn kafka
+  "Returns a function that accepts an event and sends it to a Kafka topic.
```
Comment (mcorbin) [inline src/riemann/kafka.clj]: Maybe accepts a list of events too ?

File: src/riemann/kafka.clj [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  Example:
+
+  (kafka
+    {:bootstrap-servers \"kafka.example.com:9092\"
+     :topic \"alert\"})"
+  ([] (kafka {}))
+  ([opts]
+  (let
+    [opts (merge {:bootstrap-servers "localhost:9092"
+                  :topic "riemann"
```
Comment (mcorbin) [inline src/riemann/kafka.clj]: I think it can be good to set the topic per event.

**Comment by mcorbin** [comment]
Hello,

i made some comments. 

It is possible to add optional sasl/ssl config ?

**Comment by boernd** [comment]
Hi, @mcorbin thx for the feedback. I made some changes, would be great if you could review again :)

Regarding sequence of events: If you pass a sequence of events atm it would e.g. result in 
```
[{
	"service": "firstservice",
	"host": "myhost",
	"metric": 1.0,
	"state": "critical",
	"time": 1.484579601573E9
},
{
	"service": "secondservice",
	"host": "myhost",
	"metric": 2.0,
	"state": "warning",
	"time": 1.484579601573E9
}]
```
on the broker side.

Regarding SASL/SSL support: The kafka producer config properties are passed 1:1 to the kafka client library. I just tested SSL. I put the files necessary for the SSL test into the test/data/kafka folder. Is that ok?

File: src/riemann/kafka.clj [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  ([] (kafka {}))
+  ([opts]
+   (let [opts (merge {:bootstrap.servers "localhost:9092"
+                      :value.serializer client/json-serializer}
+                     opts)
+         producer (client/producer (dissoc opts :value.serializer)
+                                   (:value.serializer opts))]
+     (fn make-stream [& args]
+       (fn stream [event]
+         (let [topic (first args)
```
Comment (mcorbin) [inline src/riemann/kafka.clj]: You can do using destructuring :

```clojure
(let [[topic message-key] args]
```

File: test/riemann/kafka_test.clj [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+; ssl.truststore.password=test1234
+
+(deftest ^:kafka ^:integration kafka-integration-plain-test
+  (let [consumer (client/consumer {:bootstrap.servers "localhost:9092"
+                            :group.id "plaingroup"}
+                           (client/keyword-deserializer)
+                           (client/json-deserializer))
+        producer (kafka)
+        kafka-output (producer "riemann" "mykey")
+        ]
```
Comment (mcorbin) [inline test/riemann/kafka_test.clj]: Should be at the end of the previous line

File: test/riemann/kafka_test.clj [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+; listeners=PLAINTEXT://localhost:9092,SSL://localhost:9093
+; ssl.keystore.location=<path_to_riemann>/riemann/test/data/kafka/kafka.server.keystore.jks
+; ssl.keystore.password=test1234
+; ssl.key.password=test1234
+; ssl.truststore.location=<path_to_riemann>/riemann/test/data/kafka/kafka.server.truststore.jks
+; ssl.truststore.password=test1234
+
+(deftest ^:kafka ^:integration kafka-integration-plain-test
+  (let [consumer (client/consumer {:bootstrap.servers "localhost:9092"
+                            :group.id "plaingroup"}
```
Comment (mcorbin) [inline test/riemann/kafka_test.clj]: Should be indented at the :bootstrap.servers level.

File: test/riemann/kafka_test.clj [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+        (is (= "myhost" 
+               (get-in (last (get-in result [:by-topic "riemann"])) [:value :host])))
+        (is (= 1.0 
+               (get-in (last (get-in result [:by-topic "riemann"])) [:value :metric])))
+        (is (= "critical" 
+               (get-in (last (get-in result [:by-topic "riemann"])) [:value :state])))))))
+
+(deftest ^:kafka ^:integration kafka-integration-ssl-test
+  (let [consumer (client/consumer {:bootstrap.servers "localhost:9093"
+                            :security.protocol "SSL"
```
Comment (mcorbin) [inline test/riemann/kafka_test.clj]: Should be indented at the :bootstrap.servers level.

File: test/riemann/kafka_test.clj [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+                            :ssl.truststore.location "test/data/kafka/kafka.client.truststore.jks"
+                            :ssl.truststore.password "test1234"
+                            :group.id "sslgroup"}
+                           (client/keyword-deserializer)
+                           (client/edn-deserializer))
+        producer (kafka {:bootstrap.servers "localhost:9093"
+                  :security.protocol "SSL"
```
Comment (mcorbin) [inline test/riemann/kafka_test.clj]: Should be indented at the :bootstrap.servers level.

File: test/riemann/kafka_test.clj [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+  (:use riemann.kafka
+        [riemann.time :only [unix-time]]
+        clojure.test)
+  (:require [kinsky.client :as client]
+            [riemann.logging :as logging]))
+
+(logging/init)
+
+(def ^:const first-event {:service "firstservice"
+                         :host "myhost"
```
Comment (mcorbin) [inline test/riemann/kafka_test.clj]: Should be indented at the :service level.

File: test/riemann/kafka_test.clj [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+                         :metric 1.0
+                         :state "critical"
+                         :time (unix-time)})
+
+(def ^:const second-event {:service "secondservice"
+                         :host "myhost"
```
Comment (mcorbin) [inline test/riemann/kafka_test.clj]: Should be indented at the :service level.

**Comment by mcorbin** [comment]
Hi @boernd,

I made a few comments, mostly indentation. Otherwise, it's OK for me. I will let @pyr take a look ;)

**Comment by pyr** [comment]
Thanks @boernd !

**Pull request merged**